File: SchoolManagement/school-management-next/src/types/teacher.ts
export interface Teacher {
  teacherId: string;
  name: string;
  subject: string;
  email: string;
  phoneNumber: string;
  address: string;
  classIds: string[] | null;
}

export interface RegisterTeacherPayload {
  name: string;
  teacherId: string;
  subject: string;
  email: string;
  phoneNumber: string;
  address: string;
}

export interface UpdateTeacherPayload {
  teacherId: string;
  name?: string;
  subject?: string;
  email?: string;
  phoneNumber?: string;
  address?: string;
}

export interface AssignTeacherToClassPayload {
  teacherId: string;
  classId: string;
}

export interface RemoveTeacherFromClassPayload {
  teacherId: string;
  classId: string;
}

File: SchoolManagement/SchoolManagement.Domain/Aggregates/Teachers/Payloads/Teacher.cs
using Sekiban.Pure.Aggregates;

namespace SchoolManagement.Domain;

[GenerateSerializer]
public record Teacher(
    string Name,
    string TeacherId,
    string Email,
    string PhoneNumber,
    string Address,
    string Subject,
    List<Guid> ClassIds = null
) : IAggregatePayload
{
    public Teacher() : this("", "", "", "", "", "", new List<Guid>()) { }
    
    public List<Guid> ClassIds { get; init; } = ClassIds ?? new List<Guid>();
}